### PR TITLE
Add Graviton3 instance as Jenkins CI agent

### PR DESCRIPTION
**Pull request by Liam-Sturge**
Modified the Terraform vars file to add a new Graviton3 agent which has SVE support.

cc @areusch @tqchen @driazati @leandron

### CODE REVIEW

**Review by areusch — APPROVED** [review]

File: terraform/vars/tvm-ci-prod.auto.tfvars [hunk 1]
```diff
@@ -119,6 +119,15 @@ autoscaler_types = {
     on_demand_percentage_above_base_capacity = 0
     on_demand_base_capacity                  = 0
   }
+  "Prod-Autoscaler-Jenkins-ARM-Graviton3" = {
+    image_family                             = "jenkins-stock-agent-arm"
+    agent_instance_type                      = "c7g.large"
+    labels                                   = "ARM-GRAVITON3"
```
Comment (driazati) [inline terraform/vars/tvm-ci-prod.auto.tfvars:125]: Can you also add an `ARM-GRAVITON3-SPOT` type as well that follows the pattern of the others? For the CI we've transitioned to using AWS Spot runners to save costs, falling back to on-demand only when those get terminated during a job.

**Review by driazati — COMMENTED** [review]
Code LGTM. If we have a Graviton3 instance running ARM tests do we really need to also have the `m6g.4xlarge` and related `ARM` instances still or can we consolidate on Graviton3? CI cost is a concern so I wouldn't want us to add a new instance type that is running and testing 90% of the same thing as another

**Comment by Liam-Sturge** [comment]
Thanks for merging this @driazati . What are the next steps required to use these nodes in Jenkins? I assume the Terraform script needs to run in order to update Jenkins, but I'm unsure if this happens automatically or requires a restart of the server. Is this something you can help me with?

**Comment by driazati** [comment]
This workflow has to run and succeed to set everything up in AWS: https://github.com/tlc-pack/ci/actions/workflows/terraform_apply.yml. Some things Jenkins can do without restarting, including updating the types of nodes available, so once that is done this workflow needs to run to update Jenkins: https://github.com/tlc-pack/ci/actions/workflows/jenkins_prepare.yml

**Comment by Liam-Sturge** [comment]
OK, thanks for explaining. It looks like the terraform_apply workflow ran successfully, so we can proceed. I lack permissions to run the jenkins_prepare workflow myself. Please could you trigger this for us.

**Comment by Liam-Sturge** [comment]
@driazati @areusch, it seems that the jenkins_prepare workflow is failing and this is preventing us from being able to use the Graviton3 nodes.
See https://github.com/tlc-pack/ci/actions/runs/5057219816/jobs/9075702080

**Pull request merged**